### ISSUE TITLE
Again accented OSX path problems

### ISSUE THREAD

**Issue by juanfal** [SublimeText/LaTeXTools#579]
It is unable to build something inside:

    "/Users/me/Documents/Departamento/DOCENCIA/P1-TT/PRÁCTICAS/201415/02.Pract02.tex"'

it seems decomposed UTF8 mac file names doesn't like Python or SublimeText or both.

    Latexmk: applying rule 'pdflatex'...
    ------------
    Running '/Library/TeX/texbin/pdflatex  -recorder  -file-line-error -max-print-line=200 -synctex=1 "/Users/me/Documents/Departamento/DOCENCIA/P1-TT/PRÁCTICAS/201415/02.Pract02.tex"'
    ------------
    /Library/TeX/texbin/pdflatex: unrecognized option `-max-print-line=200'

but the Á accent is seem as decomposed in ST3:

<img width="327" alt="captura de pantalla 2015-11-08 a las 21 20 27" src="https://cloud.githubusercontent.com/assets/895596/11022415/9c25e0aa-865e-11e5-9f34-61a56f242b3d.png">

**Comment by ig0774** [comment]
Your problem with building doesn't seem related to the accented character. You've  configured the builder to pass a `max-print-line` option to pdflatex on OS X. `-max-print-line` as a command-line option is only supported on MiKTeX, so you need to do something like this: http://tex.stackexchange.com/questions/83600/how-can-i-use-max-print-line-in-pdftex-texlive (i.e., use an environment variable or modify `texmf.cnf` rather than using a command line option).

In the vanilla version of the plugin, you could change the `builder_settings` `osx` block to something like this to get the desired effect:

```json
"builder_settings": {
    "osx": {
        "env": {
             "max-print-line": 200
        }
    }
}
```

However, from your output, I suspect you're not using this plugin at all. Or are using an old version of it from before the mbuilder code was merged... Certainly from a quick test, the plugin has no problem building a document in a folder `/Users/me/áhola!`.

**Comment by juanfal** [comment]
Thanks for your quick answer.  There must be something different here.  Consider that when I "clean" the path of the file.tex from accented characters (or put it in a clean-from-accented-chars path) it works like a charm.  It seems the accented chars, which on the other hand, are not recognised by SublimeTex2/3, are sent wrongly encoded to latexmk and it misleads the interpretation of the rest of the command line.  ST is the only editor on OSX that doesn't cope well with decomposed utf8 Mac file strategy.  I can't imagine why.  When you copy a path from Finder and paste it inside a ST doc, it always breaks the accents.  I need to convert the pasteboard before pasting it in ST:

    pbpaste | iconv -f  utf-8-mac -t utf-8 | pbcopy

What I don't understand is why suddenly this problem (solved some time ago) arises again.

**Comment by ig0774** [comment]
Yes, there is something different. I was considering the case of using the (newish) mbuilder system (I think from the release last March?) which behaves a bit differently. If I switch to using the old builder, I get something like the behaviour you're mentioning, though not a failure to build per se. My `.sublime-build` I'm using looks like this:

```json
{
	"cmd": ["/Library/TeX/texbin/latexmk", 
	"-e", "\\$pdflatex = '/Library/TeX/texbin/pdflatex %O -file-line-error --max-print-line=200 -synctex=1 %S'",
	"-e", "\\$pdf_previewer = qq{if [ -n \"`ps axww | grep Skim | grep -v grep`\" ] ; then osascript -e 'tell app \"Skim\" to revert front document' ; else open -a Skim %O %S ; fi}",
	"-pv", "-pdf",  "$file"],
	"working_dir": "$file_path", /* crucial or latexmk will spew stuff in /!!! */
	"file_regex": "^(...*?):([0-9]+): ([0-9]*)([^\\.]+)",
	"selector": "text.tex.latex"
}
```

I can't get it not to build, but I do see the decomposed output in the build console:

```
Latexmk: This is Latexmk, John Collins, 5 February 2015, version: 4.43a.
Rule 'pdflatex': Rules & subrules not known to be previously run:
   pdflatex
Rule 'pdflatex': The following rules & subrules became out-of-date:
      'pdflatex'
------------
Run number 1 of rule 'pdflatex'
------------
------------
Running '/Library/TeX/texbin/pdflatex  -recorder  -file-line-error --max-print-line=200 -synctex=1 "/Users/me/a´loha!/test.tex"'
------------
Latexmk: applying rule 'pdflatex'...
/Library/TeX/texbin/pdflatex: unrecognized option `--max-print-line=200'
...
```

Nevertheless, the PDF file is created and it opens in Skim.app. You may want to compare your `.sublime-build` to the above to see where things are different (if you can post your `.sublime-build` file that might help). It may also be that I am using a newer version of Latexmk or TeXLive than you which handles the decomposed output path properly. Those are the two possible things that occur to me.

On the other hand, its probably a better idea to switch to the post-2014 build system, since we have greater programmatic control over what happens there inside the plugin. There's nothing that the plugin itself can do to get ST to properly support the decomposed accents on OS X; that has to be addressed within the ST codebase itself. With the new build system, however, the building is all done in Python, and while I'm not certain its 100% there in terms of supporting non-ascii characters, we can at least fix those issues as and when they arise.

**Comment by juanfal** [comment]
The output for a file named: `'/Users/me/Desktop/año/año.tex'`

    Latexmk: This is Latexmk, John Collins, 5 February 2015, version: 4.43a.
    Rule 'pdflatex': File changes, etc:
       Non-existent destination files:
          'año.pdf'
    ------------
    Run number 1 of rule 'pdflatex'
    ------------
    ------------
    Running '/Library/TeX/texbin/pdflatex  -recorder  -file-line-error --max-print-line=200 -synctex=1 "/Users/juanfc/Desktop/año/año.tex"'
    ------------
    Latexmk: applying rule 'pdflatex'...
    /Library/TeX/texbin/pdflatex: unrecognized option `--max-print-line=200'
    This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015) (preloaded format=pdflatexs12)
     restricted \write18 enabled.
    entering extended mode
    ! I can't find file `/Users/me/Desktop/an'.
    <to be read again> 
                       \begingroup 
    <*> /Users/me/Desktop/an^^cc^^83
                                        o/an^^cc^^83o.tex
    (Press Enter to retry, or Control-D to exit)
    Please type another input file name: 

My .sublime.build is the one that comes with the last fresh new installed LaTeXTools

How could you reencoded a Python badly encoded string?

**Comment by ig0774** [comment]
So the latest `.sublime-build` doesn't produce output like what your showing. It produces something like this:

```
[Compiling /Users/me/Desktop/año/á.tex]

TraditionalBuilder: Engine: pdflatex -> pdf. Invoking latexmk... done.

No errors.



[Done!]
```

The `.sublime-build` file you are using presumably looks something like [the version in my last comment](https://github.com/SublimeText/LaTeXTools/issues/579#issuecomment-155424649). In this instance it's ST which creates in the command line, by plugging values into the variables in the `cmd` setting.

In the latest `.sublime-build`, instead we tell it to run the `make_pdfCommand`, a Python class and,  ultimately, from there, `latexmk` is launched using a subprocess rather than the build-in ST launching method. This is what I meant when I said we could possibly deal with encoding issues if they occur: LaTeXTools has more control over the eventual command passed to the OS using the new builder system.

In any case, the new builder could resolve the issue if it is related solely to accents in the path to the file. For example, instead of running `pdflatex /full/path/to/file`, we run `pdflatex filename` in the directory containing the file. (It's somewhat documented [here](https://github.com/SublimeText/LaTeXTools#non-ascii-characters-and-spaces-in-path-and-file-names))

OTOH, it appears from your latest example output that your *file name* contains non-ascii characters. I really can't be sure that that can be supported, as utf-8 file name support is sketchy (at best) in TeXLive itself, which is to say you'd run into the same problems running `latexmk` from the terminal.

I hope that helps...

**Comment by juanfal** [comment]
Well, you're right.  I had the older build engine.  I have voided my User/LaTexTool.sublime-build and leave the original that says it's using `make_pdf` py command.

<img width="568" alt="captura de pantalla 2015-11-13 a las 9 53 21" src="https://cloud.githubusercontent.com/assets/895596/11142292/776e260a-89ec-11e5-8ef0-35d1f4326fa0.png">

I must be very careful with accented texfileNames either. Although it doesn't issue any errors, it doesn't build, for example `año.pdf` output.  Log, reading the file, says:

<img width="307" alt="captura de pantalla 2015-11-13 a las 9 56 24" src="https://cloud.githubusercontent.com/assets/895596/11142354/f7c2e746-89ec-11e5-8c14-a3d40955917f.png">

The contents of the file, after manually opening it, 

    This is pdfTeX, Version 3.14159265-2.6-1.40.16 (TeX Live 2015) (preloaded format=pdflatexs12 2015.9.20)  13 NOV 2015 09:50
    entering extended mode
     restricted \write18 enabled.
     %&-line parsing enabled.
    **an^^cc^^83o.tex -recorder

    ! Emergency stop.
    <to be read again> 
                       \begingroup 
    <*> an^^cc^^83
                  o.tex -recorder
    *** (job aborted, file error in nonstop mode)

     
    Here is how much of TeX's memory you used:
     4 strings out of 482249
     105 string characters out of 5985729
     244375 words of memory out of 5000000
     14264 multiletter control sequences out of 15000+600000
     3938 words of font info for 15 fonts, out of 8000000 for 9000
     1141 hyphenation exceptions out of 8191
     2i,0n,1p,30b,6s stack positions out of 5000i,500n,10000p,200000b,80000s
    !  ==> Fatal error occurred, no output PDF file produced!

It doesn't compile on Terminal with `latexmk año.tex`, which issues exactly the same log, as expected.

I suppose the problem, of course, is there.  Thanks for your great work!!

**Comment by msiniscalchi** [comment]
tl;dr version: non-ASCII path/file names are iffy at best in TeX. Over the years, LaTeXTools has added workarounds to improve the situation *a bit*, but there is no perfect solution. Try switching over to a recent version of the plugin, using the builder code. Note that, thanks to recent contributed code, you can now pass options to the tex engine, so if that's the reason you kept using the sublime-build facility, it's no longer needed.

If you are satsified that things are OK (or OKish....), please close.

**Comment by juanfal** [comment]
Yes.  It works for **paths** with utf, but not for files with utf; that is, it would work for `/años/aqui.tex` but not for `/aqui/años.tex`

I understand that's not a LaTeXTools issue, but LaTeX's

Thanks
